R/db.R

```diff
@@ -37,13 +37,13 @@ st_read_db = function(conn = NULL, table, query = paste("select * from ", table,
 			SRID = dbGetQuery(conn, paste0("select srid from geometry_columns where f_table_name = '", table, "';"))[[1]]
 			ret = dbGetQuery(conn, paste("select proj4text from spatial_ref_sys where srid =", SRID, ";"))
 			if (nrow(ret))
-				ret[[1]]
+				SRID
```

Comment:
try returning `make_crs(ret[[1]])`, this should work in both cases (GDAL will try to resolve epsg from the proj4string, but leave it NA if it cannot find it). Also try the conditional example in st_read_db for an example.